### PR TITLE
Change Resources list filter from 'OR' to 'AND'

### PR DESCRIPTION
**Pull request by JackRyan1989**
[Ticket](https://github.com/usdoj-crt/beta.ADA.gov-Project-Management/issues/515)

## What does this change?

The filter on the Resources page currently operates as an 'OR' filter, meaning that it looks for items with a tag or any other tag in the filter. This can yield slightly confusing results where the filter returns a whole bunch of stuff. This change makes the filter into an 'AND' filter. Meaning that if multiple tags are selected, it finds items that match all of the tags.

Other changes: 
1. Tags are being put into data attributes rather than in the class list. This will prevent any future collisions.
2. Playwright tests have been updated.

## Screenshots (for front-end PR):

Old:
Screen Shot 2022-07-26 at 1 56 57 PM

New:
Screen Shot 2022-07-26 at 1 57 40 PM

## Checklist:

+ [x] Confirm that all items appear when selecting ONLY one of the Organization Checkboxes
+ [x] Confirm that all items appear when selecting BOTH of the Organization Checkboxes
+ [x] Confirm that 1 item appears when selecting the Artificial Intelligence Checkbox
+ [x] Do other combinations of the checkboxes and confirm by looking at the data attributes of each list item
+ [x] Confirm that the number of items found text matches the number of articles listed

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.

### CODE REVIEW

**Review by chinelo-ikejimba — APPROVED** [review]
@JackRyan1989 works as expected with the checklist combinations you wrote as criteria. Well done!

**Review by eileen-nava — APPROVED** [review]
Good work, Jack!  I tested and this works well.

**Review by billyhunt — APPROVED** [review]
Looks good to me!

File: _assets/js/utils/taSelectors/toggleVisibility.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+      state.forEach(tag => {
+        if (!tagList.includes(tag)) {
+          element.style.display = 'none';
+          numberVisible -= 1;
         }
       })
     }
   });
-  return visibleElements.length;
+  return numberVisible;
```
[the remark this answers is not in the document]
Comment (JackRyan1989) [reply _assets/js/utils/taSelectors/toggleVisibility.js]: @billyhunt Uh, sorry man but you were looking at an outdated version.

**Pull request merged**